util.go

```diff
@@ -170,3 +171,29 @@ func Equal(a, b interface{}) bool {
 	}
 	return false
 }
+
+// Walk is a filepath.Walk helper which follows symlinks and always returns the
+// path of the file or directory that is referenced in the link.
+func Walk(root string, walkFn filepath.WalkFunc) error {
+	var symWalkFn = func(path string, info os.FileInfo, err error) error {
+		var isLink bool
+		// get the target if it is a symlink
+		if err == nil && info.Mode()&os.ModeSymlink == os.ModeSymlink {
+			isLink = true
+			path, err = filepath.EvalSymlinks(path)
```

Comment:
consider a directory structure like 

test
├── a
│   ├── hello
│   └── hi
├── b -> a
└── hi -> a/hi

ideally the names of files one gives to walkFn should be:

test
test/a
test/a/hello
test/a/hi
test/b
test/b/hello
test/b/hi
test/hi

but this code always calls walkFn with fully resolved paths. making it less useful in the general case